File: Array/Easy/217. Contains Duplicate.go
package main

import "fmt"

func main() {

	arr := []int{1, 2, 3, 1}
	result := containsDuplicate(arr)
	fmt.Println(result)
}

func containsDuplicate(nums []int) bool {
	ansMap := make(map[int]int)

	for _, val := range nums {
		_, ok := ansMap[val]
		if ok {
			return true
		}
		ansMap[val]++
	}

	return false
}

File: Blind75/Array/217. Contains Duplicate.py
class Solution:
    def containsDuplicate(self, nums):
        dict = {}
        
        for num in nums:
            if num not in dict:
                dict[num] = 1
            else:
                return True
        
        return False

obj = Solution()
arr = [1,1,1,3,3,4,3,2,4,2]
ans = obj.containsDuplicate(arr)
print(ans)